js/components/graph/Graph.js

```diff
@@ -305,9 +309,12 @@ export class Graph extends React.Component {
   }
 
   componentDidUpdate(prevProps, prevState) {
-    if (prevState.nodesJSON !== this.state.nodesJSON) {
+    if (
```

Comment:
This change isn't good; it's bad practice to do a deep comparison of object values in `componentDidUpdate`, as that takes time linear in the size of the object.

If you're finding this if condition isn't triggering correctly, that means that some code isn't creating a new `nodesJSON` object, but instead mutating the existing object. You can find and modify the problematic code to create a copy of the `nodesJSON` state.